packages/core/src/App/Containers/SettingsModal/settings-language.jsx

```diff
@@ -1,91 +1,47 @@
 import React from 'react';
 import classNames from 'classnames';
-import PropTypes from 'prop-types';
-import { useTranslation } from 'react-i18next';
-import { Icon, Text } from '@deriv/components';
+import { Text } from '@deriv/components';
+import { useStore } from '@deriv/stores';
 import { Localize, getAllowedLanguages } from '@deriv/translations';
-import { changeLanguage } from 'Utils/Language';
-import { connect } from 'Stores/connect';
+import { LanguageLink } from 'App/Components/Routes';
 
 const isCurrentLanguage = (lang, current_language) => lang === current_language;
 
-const NonClickableLink = ({ children, lang }) => (
-    <div
-        id={`dt_settings_${lang}_button`}
-        className={classNames('settings-language__language-link', {
-            'settings-language__language-link--active': isCurrentLanguage(lang),
-        })}
-    >
-        {children}
-    </div>
-);
-
-const LanguageLink = ({ lang }) => (
-    <React.Fragment>
-        <Icon
-            icon={`IcFlag${lang.replace('_', '-')}`}
-            className={'settings-language__language-link-flag settings-language__language-flag'}
-            type={lang.replace(/(\s|_)/, '-').toLowerCase()}
-        />
-        <span
-            className={classNames('settings-language__language-name', {
-                'settings-language__language-name--active': isCurrentLanguage(lang),
-            })}
-        >
-            {getAllowedLanguages()[lang]}
-        </span>
-    </React.Fragment>
-);
-
-const LanguageSettings = ({ changeCurrentLanguage, current_language, toggleSettingsModal }) => {
-    const { i18n } = useTranslation();
+const LanguageSettings = () => {
```

Comment:
```suggestion
const LanguageSettings = observer(() => {
```
@yauheni-deriv Whenever we use the `useStore` hook, We need to wrap the component with `observer`.

Reply:
yes, sure. thank you